sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md

```diff
@@ -522,3 +522,58 @@ class Pipeline:
         return first_node.send(pipeline_request)  # type: ignore
 
 ```
+
+## Credentials
+
+### TokenCredential protocol
+
+Clients from the Azure SDK often require a `TokenCredential` instance in their constructors. A `TokenCredential` is
```

Comment:
I could be picky and say that clients that needs a AAD token require TokenCredential. We have SDKs that do not require it.

Reply:
Would it be more clear to rephrase this to "Most clients from the Azure SDK require a `TokenCredential` instance in their constructors"?

Reply:
Well, re-reading I'm fine, let's solve the discussion